dot/services_integration_test.go

```diff
@@ -39,6 +45,55 @@ func TestCreateStateService(t *testing.T) {
 	require.NotNil(t, stateSrvc)
 }
 
+func newStateServiceWOMock(t *testing.T) *state.Service {
+	t.Helper()
+
+	stateConfig := state.Config{
+		Path:      t.TempDir(),
+		LogLevel:  log.Info,
+		Telemetry: telemetry.NilClient{},
```

Comment:
nit `NoopClient` would be a better name, since it's not really `nil`

Reply:
Updated.